Stack/Opc.Ua.Core/Stack/Client/DiscoveryClient.cs

```diff
@@ -218,6 +266,67 @@ public partial class DiscoveryChannel
             return channel;
         }
 
+        /// <summary>
+        /// Creates a new transport channel that supports the ITransportWaitingConnection service contract.
+        /// </summary>
+        public static ITransportChannel Create(
+            ApplicationConfiguration configuration,
+            ITransportWaitingConnection connection,
+            EndpointConfiguration endpointConfiguration,
+            ServiceMessageContext messageContext)
+        {
+            // create a default description.
+            var endpoint = new EndpointDescription {
+                EndpointUrl = connection.EndpointUrl.ToString(),
+                SecurityMode = MessageSecurityMode.None,
+                SecurityPolicyUri = SecurityPolicies.None
+            };
+            endpoint.Server.ApplicationUri = endpoint.EndpointUrl;
+            endpoint.Server.ApplicationType = ApplicationType.DiscoveryServer;
+
+            ITransportChannel channel = CreateUaBinaryChannel(
+                configuration,
+                connection,
+                endpoint,
+                endpointConfiguration,
+                (X509Certificate2)null,
+                (X509Certificate2Collection)null,
```

Comment:
[Question] Can we never use certificates in this case? Maybe we should add a comment why it is not possible

Reply:
This is just for the discovery channel, which doesn't use security.

---
In reply to: [432397736](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432397736) [](ancestors = 432397736)

Reply:
added parameter for cert, but not for the chain. Not even sure why the cert is needed.

---
In reply to: [432920193](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432920193) [](ancestors = 432920193,432397736)